daisy_workflows/image_build/debian/build_fai.py

```diff
@@ -175,6 +177,54 @@ def main():
   logging.info('Saving %s to %s' % (disk_tar_gz, image_dest))
   utils.UploadFile(disk_tar_gz, image_dest)
 
+  # Create and upload metadata of the image and packages
+  logging.info('Creating image metadata.')
+  image = {
+      "id": uuid.uuid4(),
+      "family": "debian-10",
+      "name": "debian-10" + debian_version + "v" + build_date,
+      "version": debian_version,
+      "location": image_dest,
+      "release_date": build_date,
+      "release_time": time.time(),
+      "state": "Active",
+      "environment": "prod",
+      "packages": []
+  }
+  # Read list of guest package
+  with open("guest_package") as f:
+    guest_packages = f.read().splitlines()
+
+  for package in guest_packages:
+    cmd = "dpkg-query -W --showformat '${Package} ${Version} ${Git}\n'" \
+          + package
+    code, stdout = utils.Excute(cmd, capture_output=True)
+    if code == 0:
+      splits = stdout.decode('utf-8').split('\t\b')
+      package_name = splits[0]
+      package_version = splits[1]
+      package_commit_hash = splits[2][splits[2].rindex('/'):len(splits[2])]
```

Comment:
this is quite a bit of work, mostly because of the format you passed to dpkg-query. if you changed the format of dpkg-query, you could produce output that's easier to parse.

Reply:
This field is from "Git". The field is parse commit hash from git pull path.  For example: https://github.com/hopkiw/compute-image-packages/tree/33bed8dcff55a78a55bb589e071548fbc6b949ed

Reply:
I don't understand your reply, sorry. the link is to the whole repo at a point in time. and also we are not discussing a single field, but the lines 203 through 206 (at least) where we're doing string parsing. we control the output format, so we could produce a format that is explicitly designed to be easier to parse. for example we could use field separators instead of whitespace.

Reply:
Will using separator, and remove parsing the github link.